apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx

```diff
@@ -0,0 +1,199 @@
+import { VegaIcon, VegaIconNames } from '@vegaprotocol/ui-toolkit';
+import { useT } from '../../../lib/use-t';
+import classNames from 'classnames';
+import type { PartyVestingStats } from '@vegaprotocol/types';
+import BigNumber from 'bignumber.js';
+import { formatNumber } from '@vegaprotocol/utils';
+
+export const RewardHoarderBonus = ({
+  tiers,
+  vestingDetails,
+}: {
+  tiers:
+    | never[]
+    | {
+        minimum_quantum_balance: string;
+        reward_multiplier: string;
+      }[];
+  vestingDetails?: PartyVestingStats | null;
+}) => {
+  const t = useT();
+
+  const getUserTier = () => {
+    let userTier = 0,
+      i = 0;
+    if (!vestingDetails) return 0;
+    while (
+      i < tiers.length &&
+      new BigNumber(tiers[userTier].minimum_quantum_balance).isLessThan(
+        vestingDetails.quantumBalance
+      )
+    ) {
+      userTier = i;
+      i++;
+    }
+
+    if (
+      i === tiers.length &&
+      new BigNumber(
+        tiers[userTier].minimum_quantum_balance
+      ).isLessThanOrEqualTo(vestingDetails.quantumBalance)
+    ) {
+      userTier = i;
+    }
+
+    if (userTier > tiers.length) {
+      userTier--;
+    }
+
+    return userTier;
+  };
+  if (!tiers || tiers.length === 0) return null;
+  const userTierIndex = getUserTier();
+
+  // There is only value to compare to the tiers that covers all the user' rewards across all assets
+  const qUSD = 'qUSD';
+
+  const safeProgress = (i: number) => {
+    if (i < userTierIndex) return 100;
+    if (i > userTierIndex) return 0;
+    if (!vestingDetails) return 0;
+    const progress = vestingDetails.quantumBalance;
+    const total = tiers[i].minimum_quantum_balance;
+    if (!total || total === '0') return 0;
+    if (new BigNumber(progress).isGreaterThan(total)) return 100;
+    return new BigNumber(progress)
+      .multipliedBy(100)
+      .dividedBy(total || 1)
+      .toNumber();
+  };
+
+  const progressBarHeight = 'h-10';
+
+  return (
+    <>
+      <div className="flex flex-col gap-1 w-full">
+        <div className="flex flex-col gap-1">
+          <div
+            className="grid"
+            style={{
+              gridTemplateColumns:
+                'repeat(' + tiers.length + ', minmax(0, 1fr))',
+            }}
+          >
+            {tiers.map((tier, index) => {
+              return (
+                <div key={index} className="flex justify-end -mr-[2.95rem]">
```

Comment:
Is there no way to componentize the UI shared between the hoarder bonus and the activity streak?

Reply:
we can, but there are some type differences

Reply:
sometimes making them super configurable is not better. I will take a look

Reply:
I made the progress bar fill re-usable. the tiers are very different so the components will be different